mshadow/base.h

```diff
@@ -818,6 +818,25 @@ struct minimum {
     LOG(FATAL) << "Unknown type enum " << type;           \
   }
 
+#define MSHADOW_SINGLE_DOUBLE_TYPE_SWITCH(type, DType, ...)  \
```

Comment:
I feel the name is a little too long to type.

How about "MSHADOW_SGL_DBL_TYPE_SWITCH"?